c/parallel/monitor_thread.cc

```diff
@@ -13,32 +13,47 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //------------------------------------------------------------------------------
-/* enable nice() function in unistd.h */
-#define _XOPEN_SOURCE
+#include "utils/macros.h"
+#if !DT_OS_WINDOWS
+  /* enable nice() function in unistd.h */
+  #define _XOPEN_SOURCE
+  #include <unistd.h>                   // nice
+#endif
 #include <iostream>
-#include <csignal> // std::signal
+#include <csignal>                      // std::signal, sig_atomic_t
+#include "parallel/api.h"
 #include "parallel/monitor_thread.h"
-#include "parallel/thread_worker.h"     // idle_job
 #include "progress/progress_manager.h"  // dt::progress::progress_manager
+#include "parallel/thread_worker.h"     // idle_job
 #include "utils/exceptions.h"
-#include "parallel/api.h"
-#include "utils/macros.h"
-#if !DT_OS_WINDOWS
-  #include <unistd.h>        // nice
-#endif
-
-
-namespace dt {
 
+// volatile std::sig_atomic_t gSignalStatus;
 using sig_handler_t = void(*)(int);
 static sig_handler_t sigint_handler_prev = nullptr;
 
+static volatile std::sig_atomic_t monitor_thread_active = 0;
+
+extern "C" {
+  static void sigint_handler(int signal) {
+    if (monitor_thread_active) {
+      dt::progress::manager->set_interrupt();
+    }
+    else if (sigint_handler_prev) {
```

Comment:
Any ideas what was causing the segfault? This check was missing?

Reply:
Unlikely, I put it there just in case.